portal/src/main/webapp/js/src/study-view/component/PieChart.js

```diff
@@ -1013,17 +1021,14 @@ var PieChart = function(){
             }
         }
 
-        _tableDiv += '</tbody></table>';
+        _tableDiv.push('</tbody></table>');
+        _tableDivStr = _tableDiv.join('');
 
-        var pieLabelTable = $('#' + DIV.mainDiv)
-            .find('.study-view-pie-label')
-            .append(_tableDiv);
+        d3.selectAll('#' + DIV.mainDiv).selectAll('.study-view-pie-label').html(_tableDivStr);
 
-        pieLabelTable.find('table').attr('id', DIV.labelTableID+'-0');
+        $('#' + DIV.mainDiv + ' .study-view-pie-label table').attr('id', DIV.labelTableID+'-0');
```

Comment:
I mean, in line 1029 you write:
```
$('#' + DIV.mainDiv + ' .study-view-pie-label table')
```
and on line 1027, 1031 yo write: 
```
d3.selectAll('#' + DIV.mainDiv).selectAll('.study-view-pie-label-copy')
```
two different ways for selecting. Makes the code a bit messy.